aequilibrae/project/network/network.py

```diff
@@ -173,24 +199,37 @@ def create_from_osm(
         self.builder = OSMBuilder(self.downloader.json, self.source)
         self.builder.doWork()
 
-        if spatial_index:
-            logger.info("Adding spatial indices")
-            self.add_spatial_index()
-
         logger.info("Network built successfully")
 
-    def build_graphs(self) -> None:
+    def build_graphs(self, fields: list = None) -> None:
         """Builds graphs for all modes currently available in the model
 
         When called, it overwrites all graphs previously created and stored in the networks'
         dictionary of graphs
+
+        Args:
+            *fields* (:obj:`list`, optional): When working with very large graphs with large number of fields in the
+                                              database, it may be useful to specify which fields to use
+
+        To use the *fields* parameter, a minimalistic option is the following
+        ::
+
+            p = Project()
+            p.open(nm)
+            fields = ['link_id', 'a_node', 'b_node', 'direction', 'distance', 'modes']
+            p.network.build_graphs(fields)
+
         """
         curr = self.conn.cursor()
-        curr.execute('PRAGMA table_info(links);')
-        field_names = curr.fetchall()
 
-        ignore_fields = ['ogc_fid', 'geometry']
-        all_fields = [f[1] for f in field_names if f[1] not in ignore_fields]
+        if fields is None:
+            curr.execute('PRAGMA table_info(links);')
+            field_names = curr.fetchall()
+
+            ignore_fields = ['ogc_fid', 'geometry']
+            all_fields = [f[1] for f in field_names if f[1] not in ignore_fields]
+        else:
+            all_fields = fields
```

Comment:
```suggestion
        if fields is None:
            curr.execute('PRAGMA table_info(links);')
            field_names = curr.fetchall()

            ignore_fields = ['ogc_fid', 'geometry']
            fields = [f[1] for f in field_names if f[1] not in ignore_fields]
```

and then just use fields in the rest of the method?